pkg/announce/github_page.go

```diff
@@ -291,19 +292,39 @@ func UpdateGitHubPage(opts *GitHubPageOptions) (err error) {
 		return errors.New("tag not found while trying to publish release page")
 	}
 
-	// Get the release we are looking for
-	releases, err := gh.Releases(opts.Owner, opts.Repo, true)
+	// Get the release we are looking for. We only need to fetch prereleases
+	// if the release is a prerelease. If we don't filter them out, comparing
+	// e.g. v1.30.0-rc.0 with v1.29.4 will incorrectly determine that v1.29.4
+	// is *not* the latest (stable) release.
+	releases, err := gh.Releases(opts.Owner, opts.Repo, isPrerelease)
 	if err != nil {
 		return fmt.Errorf("listing the repositories releases: %w", err)
 	}
 
-	// Does the release exist yet?
+	// Does the release exist yet and should it be marked as latest?
 	var releaseID int64
 	commitish := ""
+	// No pre-release should ever be marked as "latest"
+	markAsLatest := !isPrerelease
+
 	for _, release := range releases {
 		if release.GetTagName() == opts.Tag {
 			releaseID = release.GetID()
 			commitish = release.GetTargetCommitish()
+		} else if markAsLatest {
```

Comment:
Why do we need `else` here? Running this when updating the release page should also work and it can be useful if we end up in that situation

Reply:
Ignore the comment, it's actually not important. Even if we would enter this branch when `release.GetTagName() == opts.Tag`, `semver.LE(releaseSemver)` is for sure not going to be satisfied because we're comparing the same release

Reply:
I'm not sure it's strictly needed, but I added it because then we skip a comparison of "v1.29.4" with "v1.29.4" (because we already know we are operating on the same version as the one we want to update).

Whether v1.29.4 already is marked "latest" or not doesn't matter. In both cases, `markAsLatest` will stay true (because we are not entering this part of the if/else condition) and the loop won't find any other "higher" release that will cause `markAsLatest` to become `false`. So the release will be updated and that request will include the operation to mark it as latest.

If there is a release that's newer, it cannot be the release we're currently updating, so the loop will hit that in the `else` branch and set `markAsLatest` to `false`.

Pagination is a theoretical problem but it seems the code to update an existing release is already working under the assumption that an existing release is on the first page, and "newer" releases are hopefully always on the first page.